File: Leetcode100/leetcode100_python/CountingBits2.py
from typing import List

class Solution:
    def countBits(self, num: int) -> List[int]:
        bits = [0] * (num + 1)
        i, b = 0, 1
        while b <= num:
            while i < b and i + b <= num:
                bits[i + b] = bits[i] + 1
                i += 1
            i = 0
            b <<= 1
        return bits

if __name__ == "__main__":
    num = int(input())
    solution = Solution()
    print(solution.countBits(num))

File: Leetcode100/leetcode100/src/main/java/com/jiang/leetcode100/counting_bits/CountingBits2.java
package main.java.com.jiang.leetcode100.counting_bits;

import java.util.Scanner;

class Solution2 {
    public int[] countBits(int num) {
        int[] bits = new int[num + 1];
        for(int i = 0; i <= num; i++) {
            int count = 0, n = i;
            while(n > 0) {
                n &= n - 1;
                count++;
            }
            bits[i] = count;
        }
        return bits;
    }
}

public class CountingBits2 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int num = scanner.nextInt();
        scanner.close();
        int[] result = new Solution2().countBits(num);
        for(int n : result) {
            System.out.print(n + " ");
        }
    }
}